sdk/python/kfp/v2/dsl/component_bridge.py

```diff
@@ -135,10 +127,10 @@ def _output_parameter_placeholder(output_key: str) -> str:
     return "{{{{$.outputs.parameters['{}'].output_file}}}}".format(output_key)
 
   def _resolve_output_path_placeholder(output_key: str) -> str:
-    if type_utils.is_parameter_type(outputs_dict[output_key].type):
-      return _output_parameter_placeholder(output_key)
-    else:
+    if type_utils.is_artifact_type(outputs_dict[output_key].type):
```

Comment:
JFYI: TypeSpec is not always a string, it can also be a dict such as `{'JsonObject': {'data_type': 'proto:tfx.components.trainer.TrainArgs'}}`.

Reply:
Added test cases for this in #4689